### ISSUE TITLE
Autocomplete should not suggests non-existing URLs

### ISSUE THREAD

**Issue by corwinburdick** [mozilla-mobile/fenix#21265]
### Steps to reproduce

1. Try to navigate to a website that does not exist by typing the URL into the search bar (e.g. www.fenixautocompletebug.com)
2. Close the tab
3. Open a new tab and type "fenixautocomplete"

### Expected behaviour

The URL should not be filled in by Autocomplete URLs. This behavior is consistent with Firefox Desktop.

### Actual behaviour

The non-existent website shows up in the suggestions.
[Screenshot](https://imgur.com/a/fRrjd1X)

### Device name

Google Pixel 2 XL

### Android version

Android 11

### Firefox release type

Firefox

### Firefox version

92.1.1

### Additional information

This bug's impact is exacerbated by a missing feature: the ability of Firefox Desktop to remove a URL from autocomplete by either forgetting the site through History, or using the Shift+Delete shortcut in the autocomplete dropdown.

The only current ways to stop the nonexistent site from showing up in autocomplete forever are:

- Delete all autocomplete history, with no time range to allow the user to keep most of their data
- Disable the Autocomplete URLs feature entirely

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-17856)

**Comment by SoftVision-LorandJanos** [comment]
Hi @corwinburdick  ! Thank you for the report!

I can confirm the behavior is as you described.
However, I've looked into the available documentation, and there's no evidence that the expected behavior is how you described it. A previously searched word or address should trigger autocomplete the next time you type enough of it for the address/word to appear, even if the site does not exist.
As this is at the moment an expected behavior, it is not considered a bug, so I'll remove the bug label and add a feature request label.

Verified with the latest Nightly (94.0a1 - 2021-09-17T02:58:52.277509).
Device used: Xiaomi Redmi 11 Lite (Android 11).
